mits/mariadb_test.go

```diff
@@ -17,15 +17,114 @@
 	BeforeEach(func() {
-		if !testConfig.Enabled {
+		if !tests.MariaDB.Enabled {
 			Skip("Test is disabled")
 		}
 	})
+
+	It("should deploy and connect", func() {
+		orgName := testSetup.TestSpace.OrganizationName()
+		spaceName := testSetup.TestSpace.SpaceName()
+		appName := generator.PrefixedRandomName(tests.MariaDB.Class, "app")
+		serviceName := generator.PrefixedRandomName(tests.MariaDB.Class, "service")
+		securityGroupName := generator.PrefixedRandomName(tests.MariaDB.Class, "security-group")
+
+		By("pushing the test app without starting")
+		Expect(cf.Cf("push", appName, "--no-start", "-p", "assets/mariadbapp").Wait(timeouts.CFPush.Duration())).
+			To(Exit(0))
+		defer func() {
+			cf.Cf("delete", appName, "-r", "-f").Wait(testSetup.ShortTimeout())
+		}()
+		By("setting the SERVICE_NAME environment variable in the app")
+		Expect(cf.Cf("set-env", appName, "SERVICE_NAME", serviceName).Wait(testSetup.ShortTimeout())).
+			To(Exit(0))
+
+		service := mits.NewService(serviceName, serviceBrokerName, GinkgoWriter, GinkgoWriter)
+
+		params := map[string]interface{}{
+			"db": map[string]interface{}{
+				"name": generator.PrefixedRandomName(tests.MariaDB.Class, "db"),
+				"user": generator.PrefixedRandomName(tests.MariaDB.Class, "user"),
+			},
+		}
+		By("creating the service instance")
+		err := service.Create(tests.MariaDB, params, timeouts.CFCreateService.Duration())
+		Expect(err).NotTo(HaveOccurred())
+		defer service.Destroy(testSetup.ShortTimeout())
+
+		By("waiting for the service instance to become ready")
+		err = service.WaitForCreate(timeouts.CFCreateService.Duration())
+		Expect(err).NotTo(HaveOccurred())
+
+		By("binding the service instance to the app")
+		err = service.Bind(appName, testSetup.ShortTimeout())
+		Expect(err).NotTo(HaveOccurred())
+		defer service.Unbind(appName, testSetup.ShortTimeout())
+
+		By("creating and binding a security-group for the service instance")
+		credentials, err := service.Credentials(testSetup.ShortTimeout())
+		Expect(err).NotTo(HaveOccurred())
+
+		host := credentials["host"].(string)
+		port := strconv.Itoa(int(credentials["port"].(float64)))
+		hostIP, err := net.LookupIP(host)
+		Expect(err).NotTo(HaveOccurred())
+
+		securityGroup := []map[string]string{
+			{
+				"protocol":    "tcp",
+				"destination": fmt.Sprintf("%s/32", hostIP[0]),
+				"ports":       port,
+				"description": fmt.Sprintf("Allow traffic to %s", serviceName),
+			},
+		}
+		securityGroupFile, err := ioutil.TempFile("", fmt.Sprintf("%s_security_group.json", serviceName))
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(securityGroupFile.Name())
+		encoder := json.NewEncoder(securityGroupFile)
+		err = encoder.Encode(securityGroup)
+		Expect(err).NotTo(HaveOccurred())
+		securityGroupFile.Close()
+
+		workflowhelpers.AsUser(testSetup.AdminUserContext(), testSetup.ShortTimeout(), func() {
+			Expect(cf.Cf("create-security-group", securityGroupName, securityGroupFile.Name()).Wait(testSetup.ShortTimeout())).
```

Comment:
Should we split this up, so that we always run `delete-security-group` if `create-security-group` succeeds, even if `bind-security-group` fails?